airbrake/notice.py

```diff
@@ -44,38 +44,35 @@ def __init__(self, exception, params=None, session=None,
             data = _trim_data(user, ['id', 'name', 'email'])
             if not context:
                 self.context = {}
-            self.context["user"] = data
+            self.context['user'] = data
 
         self.notifier = notifier
 
-        if isinstance(exception, str):
-            error = {
-                'type': "Error",
-                'backtrace': [{'file': "N/A",
-                               'line': 1,
-                               'function': "N/A"}],
-                'message': exception,
-                'severity': severity or ErrorLevels.DEFAULT_LEVEL}
-            self.errors = [error]
+        exc_info = sys.exc_info()
 
-        if isinstance(exception, (BaseException, Exception)):
+        # First try to use an already-created Error
+        if isinstance(exception, Error):
+            self.errors = [exception.data]
+
+        # Next try to create an Error out of the stacktrace (if available).
+        elif exc_info and exc_info[0]:
+            err = Error(exc_info)
+            self.errors = [err.data]
```

Comment:
So if `exception` is not an `Error` and we have exception info available from `sys.exc_info()` then `exception` is not used? 

This seems somewhat reasonable, but I didn't know if there was opportunity to use the `exception` value to inform one the `message` keyword argument accepted by `Error`, somehow without interrupting the way `Error` uses the exc_info tuple to build its data. 

I think what I'm really trying to say is, if I did something silly/wrong like pass a string to `Notice` as the `exception` argument, then I would expect it to either 1) fail or 2) end up as the error message (or _somewhere) in airbrake.